# OpenWeather API Testing

--------------

This notebook's purpose is to test the [One Call API 3.0](https://openweathermap.org/api/one-call-3) from Open Weather API

In [7]:
# import statements

import requests
import os
import datetime as dt
from dotenv import load_dotenv, find_dotenv

In [8]:
# import local environmental variables (so API Keys are hidden from github repo)

load_dotenv(find_dotenv(), override=True)

True

In [11]:
API_KEY = os.getenv('BLINKMETRICS_OPENWEATHER_API_KEY')

To convert from city names to lat-lon we will use the [Geocoding API](https://openweathermap.org/api/geocoding-api)

In [14]:
# testing parameters for the geocoding API

city_name = 'Beirut'
country_code = 'LB'
limit = 2

geo_code_url = f'http://api.openweathermap.org/geo/1.0/direct?q={city_name},{country_code}&limit={limit}&appid={API_KEY}'

In [19]:
# calling the api and seeing the status code (if the call worked or not)

geo_response = requests.get(geo_code_url)
geo_response.status_code

200

In [25]:
# getting the result

geo_result = geo_response.json()
geo_result

[{'name': 'Beirut',
  'local_names': {'ascii': 'Beirut',
   'oc': 'Beirot',
   'bn': 'বৈরুত',
   'et': 'Beirut',
   'lv': 'Beirūta',
   'ru': 'Бейрут',
   'es': 'Beirut',
   'br': 'Beirout',
   'bs': 'Bejrut',
   'fr': 'Beyrouth',
   'fy': 'Beirût',
   'cs': 'Bejrút',
   'ku': 'Bêrût',
   'ur': 'بیروت',
   'ar': 'بيروت',
   'is': 'Beirút',
   'mk': 'Бејрут',
   'am': 'ቤይሩት',
   'lt': 'Beirutas',
   'kk': 'Бейрут',
   'mr': 'बैरूत',
   'af': 'Beiroet',
   'he': 'ביירות',
   'pl': 'Bejrut',
   'fi': 'Beirut',
   'sv': 'Beirut',
   'el': 'Βηρυτός',
   'tg': 'Бейрут',
   'hu': 'Bejrút',
   'ka': 'ბეირუთი',
   'sr': 'Бејрут',
   'bo': 'པེ་རུ་ཐེ།',
   'yi': 'ביירוט',
   'th': 'เบรุต',
   'fa': 'بیروت',
   'pt': 'Beirute',
   'it': 'Beirut',
   'de': 'Beirut',
   'uk': 'Бейрут',
   'ug': 'بېيرۇت',
   'ml': 'ബെയ്റൂത്ത്',
   'ta': 'பெய்ரூத்',
   'sk': 'Bejrút',
   'be': 'Бейрут',
   'my': 'ဘေရွတ်မြို့',
   'eu': 'Beirut',
   'eo': 'Bejruto',
   'vo': 'Bäyrut',
   'feature_name': 'Beirut',
   'd

In [27]:
# getting the lat and lon values:

lat = geo_result[0]['lat']
lon = geo_result[0]['lon']

print((lat,lon))

(33.8959203, 35.47843)


Now to try out the One Call API

In [38]:
# testing some parameters

lat = lat
lon = lon
units = 'metric'

oc_url = f'https://api.openweathermap.org/data/3.0/onecall?lat={lat}&lon={lon}&units={units}&appid={API_KEY}'

In [40]:
# calling the api and seeing the status code (if the call worked or not)

oc_response = requests.get(oc_url)
oc_response.status_code

401

Since it's returning 401, I need to figure out what is happening since we are getting an error.

In [43]:
try:
    oc_response.raise_for_status()
    
except Exception as e:
    print(e)

401 Client Error: Unauthorized for url: https://api.openweathermap.org/data/3.0/onecall?lat=33.8959203&lon=35.47843&units=metric&appid=fda55e346b98617d3e68148dc7f96d8d


The request to the API is failing due to an authorization issue, let us see how to fix it.